packages/inter-protocol/src/auction/scheduler.js

```diff
@@ -323,7 +323,12 @@ export const makeScheduler = async (
       async updateState(_newState) {
         trace('received param update', _newState);
         await null;
-        if (!nextSchedule) {
+
+        now = await E(timer).getCurrentTimestamp();
+        if (
+          !nextSchedule ||
+          TimeMath.compareAbs(nextSchedule.startTime, now) < 0
+        ) {
```

Comment:
I think with the new await that the `await null` is redundant. But I'd propose instead only awaiting the timestamp if necessary.
```suggestion
        if (
          !nextSchedule ||
          TimeMath.compareAbs(nextSchedule.startTime, await E(timer).getCurrentTimestamp()) < 0
        ) {
```
or perhaps more readably,
```suggestion
        const validSchedule = nextSchedule && TimeMath.compareAbs(nextSchedule.startTime, await E(timer).getCurrentTimestamp()) < 0;
        if (!validSchedule) {
```

Reply:
This doesn't save `now`, so I end up making the call to `getCurrentTimestamp` inside a conditional.
```
        let repairableSchedule;
        if (!nextSchedule) {
          repairableSchedule = true;
        } else {
          now = await E(timer).getCurrentTimestamp();
          repairableSchedule =
            TimeMath.compareAbs(nextSchedule.startTime, now) < 0;
        }

        if (repairableSchedule) {
```
The TimeMath comparison will only be called if we did need to update `now`, so it's relatively clean. I tried to use a ternary, but it wasn't pretty (since the else has to be two statements--it's possible with a comma expression, but not great.)

Reply:
Ah, I didn't realize `now` needs to be saved.